components/nexus-core/src/main/java/org/sonatype/nexus/internal/blobstore/BlobStoreManagerImpl.java

```diff
@@ -81,39 +119,102 @@ protected void doStop() throws Exception {
 
   @Override
   @Guarded(by = STARTED)
+  public Iterable<BlobStore> browse() {
+    return ImmutableList.copyOf(stores.values());
+  }
+
+  @Override
+  @Guarded(by = STARTED)
+  public BlobStore create(final BlobStoreConfiguration configuration) throws Exception {
+    checkNotNull(configuration);
+    log.debug("Creating blob-store: {} with content root: {}", configuration.getName(),
+        configuration.getPath());
+
+    store.create(configuration);
+
+    BlobStore blobStore = newBlobStore(configuration);
+    track(configuration.getName(), blobStore);
+    
+    blobStore.start();
+    //TODO - event publishing
+
+    return blobStore;
+  }
+
+  @Override
+  @Guarded(by = STARTED)
+  public BlobStore update(final BlobStoreConfiguration configuration) throws Exception {
+    checkNotNull(configuration);
+    String name = checkNotNull(configuration.getName());
+    String path = checkNotNull(configuration.getPath());
+    
+    log.debug("Updating BlobStore: {}, {}", name, path);
+    
+    //TODO - changing the path will leave behind previous content, but is that sensible?
```

Comment:
What do we want to do for update use case? Only thing we can really change is the path, and leaving behind any existing content doesn't make much sense...